WalletWasabi.Gui/MainWindow.xaml.cs

```diff
@@ -78,74 +87,114 @@ private async void MainWindow_ClosingAsync(object? sender, CancelEventArgs e)
 			}
 		}
 
-		private async Task ClosingAsync()
+		/// <summary>
+		/// This will try to close the main window of the application.
+		/// </summary>
+		/// <param name="tryToDequeue">If true then the user will get a dialog where a status of the dequing process is shown and the window will be kept open until it successfully finishes. If false then the dequeing process will be handled later at the disposal of the business logic.</param>
+		/// <returns></returns>
+		public async Task ClosingAsync(bool tryToDequeue)
 		{
-			bool closeApplication = false;
+			if (IsClosed)
+			{
+				return;
+			}
+
+			if (Dispatcher.UIThread?.CheckAccess() is false)
```

Comment:
```suggestion
			if (Dispatcher.UIThread.CheckAccess() is false)
```

Can `UIThread` be `null` at this point?

Reply:
Logically the View cannot exist without the UIThread. But I don't have the guarantee it is Avalonia's internal.